tools/resources/ignore.py

```diff
@@ -31,11 +31,46 @@ class MbedIgnoreSet(object):
 
     def __init__(self):
         self._ignore_patterns = []
-        self._ignore_regex = re.compile("$^")
+        self._ignore_regexes = []
+        self._unignore_patterns = []
+        self._unignore_regexes = []
 
     def is_ignored(self, file_path):
         """Check if file path is ignored by any .mbedignore thus far"""
-        return self._ignore_regex.match(normcase(file_path))
+
+        # find longest ignore and unignore pattern that matches the path
+        ignore_match_pattern = None
+        unignore_match_pattern = None
+
+        filepath_normcase = normcase(file_path)
+
+        for regex_index in range(0, len(self._ignore_regexes)):
```

Comment:
Could we use `zip` here to tidy this loop up? Then we can say something like:
```
for ignore_regex, ignore_pattern in zip(self._ignore_regexes, self._ignore_patterns):
    this_regex_match = ignore_regex.match(filepath_normcase)
    ...
```
This way we have both of the elements there without having to introduce the extra `regex_index` variable.

Reply:
heh you can tell I'm mostly a C++ programmer... I had no idea this existed